src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parser Class is responsible for interpreting the commands entered by the user. Upon invalid commands, exceptions
+ * will be thrown and handled.
+ */
+public class Parser {
+    private String command = "";
+    private String secondaryInput = "";
+
+    /**
+     * Constructor for the Parser Class.
+     */
+    public Parser() {
+
+    }
+
+    /**
+     * Method is responsible for interpreting commands send by the user and handling invalid commands.
+     * @param commandInput
+     * @param duke
+     * @param taskList
+     */
+    public String input(String commandInput, Duke duke, TaskList taskList) {
+        boolean isCommandValid = false;
+        this.command = "";
+        this.secondaryInput = "";
+
+        if (commandInput.equals("save")) {

```

Comment:
Is there a way to keep track of loose strings like "save" in a constant or enum? A central area to store such strings as constants can make a large codebase more readable. The benefits are as follows
1. can go to a command enum file or command constants to see all 
2. can change all instances of the same "save" command, by changing the string saved in a constant or enum to "update"